src/box/iproto.cc

```diff
@@ -1003,6 +1027,8 @@ iproto_msg_start_processing_in_stream(struct iproto_msg *msg)
 	 * the message only if the message queue in the stream was empty.
 	 */
 	bool was_not_empty = !stailq_empty(&stream->pending_requests);
+	con->iproto_thread->requests_in_stream_queue += (was_not_empty ? 1 : 0);
```

Comment:
because all time when this function return 0 we push message in tx thread and we should -1 this value

Reply:
Can you rework the queue so that it only stores messages waiting to be processed, not all stream messages? Then the counter would be consistent with the queue length.

Reply:
fixed